hedera-node/hedera-addressbook-service/src/main/java/com/hedera/node/app/service/addressbook/AddressBookHelper.java

```diff
@@ -52,11 +52,14 @@ public class AddressBookHelper {
      */
     public static long getNextNodeID(@NonNull final ReadableNodeStore nodeStore) {
         requireNonNull(nodeStore);
-        final long maxNodeId = StreamSupport.stream(
-                        Spliterators.spliterator(nodeStore.keys(), nodeStore.sizeOfState(), DISTINCT), false)
-                .mapToLong(EntityNumber::number)
-                .max()
-                .orElse(-1L);
+        long maxNodeId = -1L;
+        if (nodeStore.keys() != null) {
```

Comment:
When will this be null ?

Reply:
It's good to add null check here because Spliterators.spliterator() check for requireNonNull(iterator). If it's a mock object such as in FreezeHandlerTest#happyPathFreezeUpgradeOrTelemetryUpgrade, we get nullpointerexception.

Reply:
We do not need to fix this in this PR, but I think this is not good practice. `ReadableNodeStore.keys()` should be annotated with `NonNull`, and tests need to be adjusted to behave like the real code.

Reply:
Agreed, we should adjust test in later PR.  It might be confusing in the future when we add null checks, when not required.

Reply:
I will update the test and remove null check in the next PR.